ui/ui-components/hooks/useApi.ts

```diff
@@ -18,7 +18,7 @@ export interface ApiHook {
 }
 
 export function UseApi(
-  ctx: GetServerSidePropsContext | NextPageContext,
```

Comment:
This change may benefit from refactoring useApi first. I'm not familiar with next but it seems that the context is only needed on the server-side requests. Once the page renders and uses this "hook", does it need to also pass a context from the props? Where does it come from?

Reply:
Client side it won't have the context.

Reply:
Right! So It does make sense to rebuild `useAPI` as a proper hook or a server-side utility. That will be done separately from this PR.